### ISSUE TITLE
E1101 raised when accessing a not annotated property returning a dynamic object

### ISSUE THREAD

**Issue by sevdog** [pylint-dev/pylint#9789]
### Bug description

```python
from django_auth_ldap.backend import LDAPBackend as _LDAPBackend


class LDAPBackend(_LDAPBackend):
    def my_method(self, *args, **kwargs):
        self.settings.PERMIT_EMPTY_PASSWORD
```

The settings is an object which dynamically sets attributes via `setattr` in the init.

See:

https://github.com/django-auth-ldap/django-auth-ldap/blob/master/django_auth_ldap/backend.py#L110-L119

https://github.com/django-auth-ldap/django-auth-ldap/blob/aa04ca5ac80ce3b3e7abbbcf860d684460907937/django_auth_ldap/config.py#L45-L96

### Command used

```shell
pylint backend.py -e E1101 -d C -d W
```

### Pylint output

```python
************* Module e1101.backend
e1101/backend.py:7:8: E1101: Instance of 'LDAPSettings' has no 'PERMIT_EMPTY_PASSWORD' member (no-member)
e1101/backend.py:7:8: E1101: Instance of 'LDAPBackend' has no 'PERMIT_EMPTY_PASSWORD' member (no-member)
```

### Expected behavior

No error should be raised

### Pylint version

```shell
pylint 3.2.5
astroid 3.2.3
python 3.10.12
```

### OS / Environment

Ubuntu 22.04 LTS

### Additional dependencies

```python
django-auth-ldap==4.8.0
django==5.0.7 # also on 4.2
```

**Comment by jacobtylerwalls** [comment]
Thanks for the report. Duplicate of #2714
